src/rules/npmNamingRule.ts

```diff
@@ -1,25 +1,177 @@
-import critic = require("dts-critic");
+import {
+    CheckOptions as CriticOptions,
+    CriticError,
+    dtsCritic as critic,
+    ErrorKind,
+    ExportErrorKind,
+    Mode,
+    NameOnlyOptions,
+    parseExportErrorKind,
+    parseMode } from "dts-critic";
 import * as Lint from "tslint";
 import * as ts from "typescript";
 
 import { failure, isMainFile } from "../util";
 
+type Options = CriticOptions & { singleLine?: boolean };
+
+const defaultOptions: NameOnlyOptions = {
+    mode: Mode.NameOnly,
+};
+
 export class Rule extends Lint.Rules.AbstractRule {
     static metadata: Lint.IRuleMetadata = {
         ruleName: "npm-naming",
         description: "Ensure that package name and DefinitelyTyped header match npm package info.",
-        optionsDescription: "Not configurable.",
-        options: null,
+        optionsDescription: `An object with a \`mode\` property should be provided.
+If \`mode\` is '${Mode.Code}', then option \`errors\` can be provided.
+\`errors\` should be an array specifying which code checks should be enabled or disabled.`,
+        options: {
+            oneOf: [
+                {
+                    type: "object",
+                    properties: {
+                        "mode": {
+                            type: "string",
+                            enum: [Mode.NameOnly],
+                        },
+                        "single-line": {
+                            description: "Whether to print error messages in a single line. Used for testing.",
+                            type: "boolean",
+                        },
+                        "required": ["mode"],
+                    },
+                },
+                {
+                    type: "object",
+                    properties: {
+                        "mode": {
+                            type: "string",
+                            enum: [Mode.Code],
+                        },
+                        "errors": {
+                            type: "array",
+                            items: {
+                                type: "array",
+                                items: [
+                                    {   description: "Name of the check.",
+                                        type: "string",
+                                        enum: [ErrorKind.NeedsExportEquals, ErrorKind.NoDefaultExport] as ExportErrorKind[],
```

Comment:
are these the only two error kinds that the lint rule should support?

Reply:
Yes, at least for now.